PurchasesTests/Attribution/AttributionPosterTests.swift

```diff
@@ -26,7 +26,13 @@ class AttributionPosterTests: XCTestCase {
     var deviceCache: MockDeviceCache!
     var identityManager: MockIdentityManager!
     var backend: MockBackend!
-    let subscriberAttributesManager = MockSubscriberAttributesManager()
+    lazy var subscriberAttributesManager: MockSubscriberAttributesManager = {
+        return MockSubscriberAttributesManager(
```

Comment:
this is somewhat dangerous: 
it means that we'll only ever get one instance of the mock. If we don't remember to reset the mock in between tests, we might have tests that check for a call count, that are getting weird values because of a previous test. 

What we've been doing for these (open to suggestions) is to have an implicitly-unwrapped property: 

```swift
var subscriberAttributesManager: MockSubscriberAttributesManager!
```

And then in the `setup`: 
```swift
subscriberAttributesManager = MockSubscriberAttributesManager(...
```

Then it gets called automatically before every test. 
If we forget to call it, tests will crash as soon as they try to use the property, so it'll be easy to find.